flutter_app/lib/main.dart

```diff
@@ -35,45 +36,9 @@ class HomePage extends StatelessWidget {
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: Text(title),
-      ),
-      body: Column(
-        children: [
-          // attitude
-          DataField<double>(
-            name: 'Yaw (deg)',
-            value: comm.getYawStream(),
-            formatter: (double value) => (value / pi * 180.0).round(),
-          ),
-          DataField<double>(
-            name: 'Pitch (deg)',
-            value: comm.getPitchStream(),
-            formatter: (double value) => (value / pi * 180.0).round(),
-          ),
-          DataField<double>(
-            name: 'Roll (deg)',
-            value: comm.getRollStream(),
-            formatter: (double value) => (value / pi * 180.0).round(),
-          ),
-
-          // global position
-          DataField<int>(
-            name: 'Latitude',
-            value: comm.getLatStream(),
-            formatter: (int value) => (value / 1e7).round(),
-          ),
-          DataField<int>(
-            name: 'Longitude',
-            value: comm.getLonStream(),
-            formatter: (int value) => (value / 1e7).round(),
-          ),
-          DataField<int>(
-            name: 'Altitude (m)',
-            value: comm.getAltStream(),
-            formatter: (int value) => (value / 1e3).round(),
-          ),
-        ],
+        title: Center(child: Text(title)),
       ),
+      body: DroneInformation(comm: comm),
```

Comment:
Wrap your widget in a `Container` or `SizedBox` with a fixed width, I think a value between 300 and 500 should look good.